src/main/java/ee/carlrobert/codegpt/settings/state/AzureSettingsState.java

```diff
@@ -5,25 +5,46 @@
 import com.intellij.openapi.components.State;
 import com.intellij.openapi.components.Storage;
 import com.intellij.util.xmlb.XmlSerializerUtil;
+import com.intellij.util.xmlb.annotations.Transient;
 import ee.carlrobert.codegpt.credentials.AzureCredentialsManager;
-import ee.carlrobert.codegpt.settings.service.ServiceSelectionForm;
+import ee.carlrobert.codegpt.settings.service.AzureServiceForm;
+import ee.carlrobert.codegpt.settings.state.util.RemoteOpenAiSettings;
+import ee.carlrobert.llm.client.openai.completion.OpenAIChatCompletionModel;
 import org.jetbrains.annotations.NotNull;
 
 @State(name = "CodeGPT_AzureSettings_210", storages = @Storage("CodeGPT_AzureSettings_210.xml"))
-public class AzureSettingsState implements PersistentStateComponent<AzureSettingsState> {
+public class AzureSettingsState extends
+    RemoteOpenAiSettings<AzureCredentialsManager> implements
```

Comment:
Hmm, do we really need to inherit two common params through multiple nested classes? I think this just adds an extra layer of complexity to the project and makes it harder to maintain. Also, keep in mind that `model` is now part of the Azure settings, which is unused.

A general rule of thumb is to prefer composition over inheritance.

Reply:
Good thing you ask, I forgot to ask, does the Azure service not allow to select a model?
https://github.com/carlrobertoh/CodeGPT/blob/c4e1f980aaa2f3330b768a7d7a53c95662d84ce5/src/main/java/ee/carlrobert/codegpt/completions/CompletionRequestService.java#L48-L56
Does the Azure API not care about providing a model?

Reply:
I removed the `model` from the `AzureSettingsState` and its corresponding form.

Reply:
> Does the Azure API not care about providing a model?

Nope, I think this is configured as part of the deployment id.

> With composition over inheritance I would agree with you if its about Classes/Services that can do things, but the state classes are only POJOs with a lot of fields and getter and setter.

Yep, but `AzureSettingsState` is not a simple POJO, it's an application-level service. Idk, maybe it's just me, but even with plain POJO, it doesn't make much sense to inherit these properties. I'd rather see it as "settings **has-a** overridden/remote settings" rather than **is-a**.